### PR TITLE
Implemented multi-tenancy support in the SqlStorage layer [tenantId column]

### PR DESCRIPTION
**Pull request by EricWittmann**
Multi-tenancy support in the SQL and KafkaSQL storages via a `tenantId` column on appropriate tables.

### CODE REVIEW

**Review by famartinrh — COMMENTED** [review]
looks good

**Review by carlesarnal — APPROVED** [review]

**Review by carlesarnal — APPROVED** [review]

File: storage/sql/src/test/java/io/apicurio/registry/storage/impl/sql/SqlRegistryStorageTest.java [hunk 1]
```diff
@@ -39,4 +68,238 @@ protected RegistryStorage storage() {
...
+        Assertions.assertEquals(rule.getConfiguration(), config.getConfiguration());
+        
+        globalRules = storage().getGlobalRules();
+        Assertions.assertNotNull(globalRules);
+        Assertions.assertFalse(globalRules.isEmpty());
+        Assertions.assertEquals(globalRules.size(), 1);
+        Assertions.assertEquals(globalRules.get(0), RuleType.COMPATIBILITY);
+    }
+    
+    public void testMultiTenant_ArtifactNotFound() throws Exception {
```
Comment (famartinrh) [inline storage/sql/src/test/java/io/apicurio/registry/storage/impl/sql/SqlRegistryStorageTest.java:243]: missing `@Test` here and in all methods below?

Comment (EricWittmann) [reply storage/sql/src/test/java/io/apicurio/registry/storage/impl/sql/SqlRegistryStorageTest.java]: whoops!  I knew it was too good to be true...

**Review by jsenko — COMMENTED** [review]
I made a comment about tenant context in the other PR https://github.com/Apicurio/apicurio-registry/pull/1111#pullrequestreview-564196023

The code looks good, but I think some of the tenant related repeated code could be reduced, e.g. `.bind(0, tenantContext.tenantId())` could be extracted to separate function(s).

**Comment by EricWittmann** [comment]
Closing this due to conflicts.  I will resolve and open a new PR.

**Pull request closed without merging**